File: backend/internal/domain/entities/course.go
package entities

import (
	"time"

	apperrors "github.com/ailms/backend/pkg/errors"
	"github.com/google/uuid"
)

type CourseStatus string

const (
	StatusDraft     CourseStatus = "draft"
	StatusPublished CourseStatus = "published"
)

type Course struct {
	ID          uuid.UUID
	OrgID       uuid.UUID
	Title       string
	Description string
	Status      CourseStatus
	CreatedBy   uuid.UUID
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

func (c *Course) Publish() error {
	if c.Status == StatusPublished {
		return apperrors.ConflictError("course is already published")
	}
	c.Status = StatusPublished
	return nil
}

File: frontend/src/types/course.ts
export type CourseStatus = 'draft' | 'published'

export interface CourseDTO {
  id: string
  title: string
  description: string
  status: CourseStatus
  org_id: string
  teacher_id: string
  created_at: string
  updated_at: string
}

export interface CreateCourseInput {
  title: string
  description: string
  status?: CourseStatus
}

export interface UpdateCourseInput {
  title?: string
  description?: string
  status?: CourseStatus
}
